pkg/health/health.go

```diff
@@ -0,0 +1,161 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package health
+
+import (
+	ctx "context"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+)
+
+func Health(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	fmt.Println("Info:  Checking if at least three mon pods are running on different nodes")
+	checkPodsOnNodes(context, operatorNamespace, clusterNamespace, "app=rook-ceph-mon", exitOnError)
+
+	fmt.Println()
+	fmt.Println(" Info:  Checking mon quorum and ceph health details")
+	checkMonQuorum(context, operatorNamespace, clusterNamespace, exitOnError)
+
+	fmt.Println()
+	fmt.Println("Info:  Checking if at least three osd pods are running on different nodes")
+	checkPodsOnNodes(context, operatorNamespace, clusterNamespace, "app=rook-ceph-osd", exitOnError)
+
+	fmt.Println()
+	CheckAllPodsStatus(context, operatorNamespace, clusterNamespace, exitOnError)
+
+	fmt.Println("Info:  checking placement group status")
+	checkPgStatus(context, operatorNamespace, clusterNamespace, exitOnError)
+
+	fmt.Println()
+	fmt.Println("Info:  checking if at least one mgr pod is running")
+	checkMgrPodsStatusAndCounts(context, operatorNamespace, clusterNamespace, exitOnError)
+}
+
+func checkPodsOnNodes(context *k8sutil.Context, operatorNamespace, clusterNamespace, label string, exitOnError bool) {
+	var daemonType string
+	if strings.Contains(label, "osd") {
+		daemonType = "osd"
+	} else if strings.Contains(label, "mon") {
+		daemonType = "mon"
+	}
+
+	opts := metav1.ListOptions{LabelSelector: label}
+	podList, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), opts)
+	if err != nil {
+		fmt.Printf("\nfailed to list %s pods with label %s: %v\n", daemonType, opts.LabelSelector, err)
+	}
+
+	var nodeList = make(map[string]string)
+	for i := range podList.Items {
+		nodeName := podList.Items[i].Spec.NodeName
+		if _, okay := nodeList[nodeName]; !okay {
+			nodeList[nodeName] = podList.Items[i].Name
+		}
+	}
+
+	if len(nodeList) < 3 {
+		fmt.Printf("\nWarning: At least three %s pods should running on different nodes", daemonType)
+	}
+
+	for i := range podList.Items {
+		fmt.Println(podList.Items[i].Name, "\t", podList.Items[i].Status.Phase, "\t", podList.Items[i].Namespace, "\t", podList.Items[i].Spec.NodeName)
+	}
+}
+
+func checkMonQuorum(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	cephHealthDetails := exec.RunCommandInOperatorPod(context, "ceph", []string{"health", "detail"}, operatorNamespace, clusterNamespace, false)
+	if cephHealthDetails == "HEALTH_OK" {
```

Comment:
`cephHealthDetails` contains a lot of status, right? Seems like this will never be equal.

Reply:
right, currently only using `strings.Contains()` but I'm planning on working locally, to use json marshaling whenever we are comparing ceph output status but will not be part of this pr since it will require changes in other commands too.

Reply:
I see, `unMarshalCephCommands()` only returns the status. Then we need to rename that method and also `cephHealthDetails` so it makes sense that it's only the status.